pandas/core/frame.py

```diff
@@ -4069,23 +4069,97 @@ def update(self, other, join='left', overwrite=True, filter_func=None,
     # ----------------------------------------------------------------------
     # Misc methods
 
+    def _get_valid_indices(self):
+        is_valid = self.count(1) > 0
+        return self.index[is_valid]
+
     def first_valid_index(self):
         """
-        Return label for first non-NA/null value
+        Return index for first non-NA/null value.
+        If all elements are non-NA/null, returns None.
+        Also returns None for empty DataFrame.
+
+        Examples
+        --------
+
+        When no null value in a DataFrame, returns first index.
+
+        >>> df = DataFrame({'A': [1, 2, 3], 'B': ['a', 'b', 'f']})
+        >>> df.first_valid_index()
+        0
+
+        When all elements in first row are null, returns second index.
+
```

Comment:
ideally would like to move these doc-strings to ``core/generic.py`` and use ``_shared_docs``. can you do that.

Reply:
@jreback 
Thank you for your advice.
Modified the doc-strings with `_shared_docs`.
Not sure to the manner of pandas doc-strings, please check it.
If it's ok, I will squash the 2nd commit to 1st one.

Note: I have no idea to share the doc-strings between `Series` and `DataFrames` with Examples, so I removed the examples.